src/app/collective-rewards/leaderboard/hooks/useGetBuildersRewards.ts

```diff
@@ -1,68 +1,144 @@
+  const {
+    data: buildersRewardsPercentage,
+    isLoading: rewardsPercentageLoading,
+    error: rewardsPercentageError,
+  } = useGetBuildersRewardPercentage(buildersAddress)
+
+  const gauges = whitelistedBuilders?.map(({ gauge }) => gauge)
+  const {
+    data: { allocationsPercentage },
+    isLoading: allocationsLoading,
+    error: allocationsError,
+  } = useGetTotalAllocations(gauges)
+
+  const {
+    data: rewardShares,
+    isLoading: rewardSharesLoading,
+    error: rewardSharesError,
+  } = useGetGaugesRewardShares(gauges)
+
+  const {
+    data: rewardsERC20,
+    isLoading: rewardsERC20Loading,
+    error: rewardsERC20Error,
+  } = useGetRewardsERC20()
 
   const {
-    data: rewardDistributedLogs,
+    data: rewardsCoinbase,
+    isLoading: rewardsCoinbaseLoading,
+    error: rewardsCoinbaseError,
+  } = useGetRewardsCoinbase()
+
+  const { cycleStartTimestamp, cycleEndTimestamp } = getPreviousCycle()
+  const {
+    data: backersClaimedEventsLastCycle,
     isLoading: logsLoading,
     error: logsError,
-  } = useGetRewardDistributedLogs(rewardToken)
-  const {
-    data: { share, projectedReward },
-    isLoading: tokenLoading,
-    error: tokenError,
-  } = useGetTokenProjectedReward(rewardToken)
-  const tokenSymbol = rewardTokenSymbol ?? ''
+  } = useGetGaugesBackerRewardsClaimed(
+    gauges,
+    undefined,
+    undefined,
+    cycleStartTimestamp.toSeconds(),
+    cycleEndTimestamp.toSeconds(),
+  )
+
+  const backersClaimedAmounts = useGetBackersClaimedRewardsAmount(gauges, backersClaimedEventsLastCycle)
 
-  const projectedRewardInHuman = Number(formatBalanceToHuman(projectedReward))
+  const isLoading =
+    rewardSharesLoading ||
+    whitelistedBuildersLoading ||
+    allocationsLoading ||
+    logsLoading ||
+    rewardsPercentageLoading ||
+    rewardsERC20Loading ||
+    rewardsCoinbaseLoading
 
-  const isLoading = buildersLoading || logsLoading || tokenLoading
-  const error = buildersError ?? logsError ?? tokenError
-  const whitelistedBuilders = builders.filter(builder => builder.status === BuilderStatusActive)
+  const error =
+    rewardSharesError ??
+    whitelistedBuildersError ??
+    allocationsError ??
+    logsError ??
+    rewardsPercentageError ??
+    rewardsERC20Error ??
+    rewardsCoinbaseError
 
   const { prices } = usePricesContext()
 
+  const priceRif = prices[RIF]?.price ?? 0
+  const priceRbtc = prices[RBTC]?.price ?? 0
+
   return {
-    data: whitelistedBuilders.map(({ address, builderName }) => {
-      const builderEvents = rewardDistributedLogs.filter(event =>
-        isAddressEqual(event.args.builder_, address),
-      )
-      const lastCycleRewards = getLastCycleRewards(builderEvents)
-      const lastCycleRewardsInHuman = Number(formatBalanceToHuman(lastCycleRewards))
+    data: whitelistedBuilders.map(({ address, builderName }, i) => {
+      const projectedRif = rewardShares && rewardsERC20 ? rewardShares[i] * rewardsERC20 : 0n
```

Comment:
We have the shares and the rewards but we also need to take into account the backer reward %